app.rb

```diff
@@ -0,0 +1,158 @@
+    @data.load_albums
+    @data.load_genres
+    @data.load_books
+    @data.load_labels
+  end
+
+  def inputs(show)
+    outputs = []
+    show.each do |s|
+      print s
+      outputs.push(gets.chomp)
+    end
+    outputs
+  end
+
+  def add_author
+    puts 'Enter first name'
+    first_name = gets.chomp
+    puts 'Enter last name'
+    last_name = gets.chomp
+    new_author = Author.new(nil, first_name, last_name)
+    @data.add_author(new_author)
+    @data.authors << new_author
+    new_author
+  end
+
+  def add_game
+    puts 'Is it multiplayer [Y / N]'
+    multiplayer = gets.chomp.downcase == 'y'
+    puts 'Enter last played at:'
+    last_played_at = gets.chomp
+    puts 'Enter the publish date in:'
+    publish_date = gets.chomp
+    new_game = Game.new(nil, publish_date, multiplayer, last_played_at)
+    puts "Enter author details\n"
+    new_author = add_author
+    new_game.add_author(new_author)
+    @data.add_game(new_game)
+    @data.games << new_game
+    puts 'Game created successfully'
+  end
+
+  def add_book
+    puts 'Enter publisher'
+    publisher = gets.chomp
+    puts 'Enter publish date'
+    publish_date = gets.chomp
+    puts 'Enter the cover state'
+    cover_state = gets.chomp
+    new_book = Book.new(nil, publish_date, publisher, cover_state)
+    puts "Enter label details\n"
+    new_label = add_label
+    new_book.add_label(new_label)
+    @data.add_book(new_book)
+    @data.books << new_book
+    puts 'Book created successfully'
+  end
+
+  def add_label
+    puts 'Enter title'
+    title = gets.chomp
+    puts 'Enter color'
+    color = gets.chomp
+    new_label = Label.new(nil, title, color)
+    @data.add_label(new_label)
+    @data.labels << new_label
+    new_label
+  end
+
+  def display_authors
+    return puts 'No authors found' if @data.authors.empty?
+
+    @data.authors.each_with_index do |author, index|
+      puts "#{index + 1}) First Name: #{author.first_name}, Last Name: #{author.last_name}"
+    end
+  end
+
+  def display_games
+    return puts 'No games found' if @data.games.empty?
+
+    @data.games.each_with_index do |game, index|
+      puts "#{index + 1}) Author: #{game.author.first_name} #{game.author.last_name}"
+      puts "Multiplayer: #{game.multiplayer}, Last played: #{game.last_played_at}, Publish date: #{game.publish_date}"
+    end
+  end
+
+  def add_album
+    puts 'Add New Album'
+    puts 'Publish Date:'
+    date = gets.chomp
+    puts 'On spotify? [Y/N]'
+    spotify = gets.chomp.upcase == 'Y'
+    new_album = MusicAlbum.new(date, spotify, nil)
+    puts "Enter Gener details\n"
+    new_genre = add_genre
+    new_album.add_genre(new_genre)
+    @data.add_album(new_album)
+    @data.albums << new_album
+    puts 'Album created successfully'
+  end
+
+  def add_genre
+    puts "Add Genre\n"
+    puts 'Genre name:'
+    genre_name = gets.chomp
+    new_genre = Genre.new(nil, genre_name)
+    @data.add_genre(new_genre)
+    @data.genres << new_genre
+    new_genre
+  end
```

Comment:
- [ ] I also came across the console error below when trying to add an album. Let's kindly fix it.

![Screenshot from 2022-11-24 23-21-02](https://user-images.githubusercontent.com/63087491/203869948-b406f94e-fbfd-4d0a-a6ad-a8782672d955.png)

Reply:
## Required
- [ ] Please address this I am still getting the same error
<img width="793" alt="Screen Shot 2022-11-25 at 11 22 47" src="https://user-images.githubusercontent.com/44745855/203946463-4322f844-451f-4253-ac19-758e16fe4092.png">